datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/manifest.py

```diff
@@ -65,22 +68,47 @@ def is_metric_in_metadata_file(metric, check):
     return False
 
 
+def get_original_manifest_field(check_name, field_to_check, repo_url, manifests_from_master):
+    repo_url = f'{repo_url}/{check_name}/manifest.json'
+    if manifests_from_master.get(check_name):
+        data = manifests_from_master.get(check_name)
+    else:
+        data = requests.get(repo_url.format(check_name))
```

Comment:
Unfortunately despite the caching (which is actually of limited impact because each check calls this function _at least_ once), the performance hit of adding these HTTP calls looks tremendous - about 1s per check. Manifest validation with `-i` now takes ~2min when it used to be < 1s. I think we should improve this because it's impractical, so leaving a note here, planning to explore after QA (I have ideas using async…).

Reply:
The main reason for this cache is because we call this method once per `FIELDS_NOT_ALLOWED_TO_CHANGE` per integration. So I was hoping to only have the http request happen at most once per integration. 

I agree there is more performance work to do here though.